src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1108,6 +1108,322 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
+   const std::string basis_name = "QF_Default_" + std::to_string(qf->GetSpace()->GetOrder())
+                                + "_" + std::to_string(qf->GetVDim());
+   v->setString(basis_name);
+
+   // Set the topology of the QuadratureFunction.
+   // This is always 'mesh'
+   v = alloc_view(grp, "topology")->setString("mesh");
+
+   // Set the View "<m_bp_grp>/fields/<field_name>/values"
+   addQuadratureFunction(field_name, qf, buffer_name, offset);
+
+
+   // Register field_name in the blueprint_index group.
+   if (myid == 0)
+   {
+      RegisterQFieldInBPIndex(field_name, qf);
+   }
+
+   // Register field_name + qf in field_map.
+   mfem::DataCollection::RegisterQField(field_name, qf);
+}
+
+// private method
+// Should only be called on mpi rank 0 ( or if serial problem ).
+void MFEMSidreDataCollection::
+DeregisterQFieldInBPIndex(const std::string& field_name)
+{
+   sidre::Group * fields_grp = m_bp_index_grp->getGroup("fields/");
+   SLIC_WARNING_IF(fields_grp->hasGroup(field_name),
+               "No field exists in blueprint index with name " << field_name);
+
+   // Note: This will destroy all orphaned views or buffer classes under this
+   // group also.  If sidre owns this field data, the memory will be deleted
+   // unless it's referenced somewhere else in sidre.
+   fields_grp->destroyGroup(field_name);
+}
+
+void MFEMSidreDataCollection::DeregisterQField(const std::string& field_name)
+{
+   // Deregister field_name from field_map.
+   mfem::DataCollection::DeregisterQField(field_name);
+
+   sidre::Group * fields_grp = m_bp_grp->getGroup("fields/");
+   SLIC_WARNING_IF(fields_grp->hasGroup(field_name),
+               "No field exists in blueprint with name " << field_name);
+
+   // Delete field_name from the blueprint group.
+
+   // Note: This will destroy all orphaned views or buffer classes under this
+   // group also.  If sidre owns this field data, the memory will be deleted
+   // unless it's referenced somewhere else in sidre.
+   fields_grp->destroyGroup(field_name);
+
+   // Delete field_name from the blueprint_index group.
+   if (myid == 0)
+   {
+      DeregisterQFieldInBPIndex(field_name);
+   }
+
+   // Delete field_name from the named_buffers group, if allocated.
+   FreeNamedBuffer(field_name);
+}
+
+int MFEMSidreDataCollection::GetQFieldOrder(const std::string &field_name)
+{
+   sidre::Group* grp = nullptr;
+   if( m_bp_grp->hasGroup("fields/" + field_name))
+   {
+      grp = m_bp_grp->getGroup("fields/" + field_name);
+
+      sidre::View *v = grp->getView("basis");
+      std::string basis_name = v->getString();
+
+      size_t pos = 0;
+      std::string delimiter = "_";
+
+      // If the DC was reloaded the internal q_field_map was not restored
+      // so we can't just use HasQField here. We should fix this in the  Load
+      // function. 
+      if (basis_name.substr(0, basis_name.find(delimiter)) != "QF")
+      {
```

Comment:
It might be useful to implement something like [mfem::FiniteElementCollection::new](https://mfem.github.io/doxygen/html/classmfem_1_1FiniteElementCollection.html#a15fcfa553d4949eb08f9926ac74d1e80) for this.  It probably belongs in MFEM but it could be temporarily kept here.